### DISCUSSION TITLE
In SSE, How can i get the details's usage in prompt(prompt_tokens, completion_tokens, total_tokens)

### DISCUSSION THREAD

**Discussion by AutumnWhj** [waylaidwanderer/node-chatgpt-api#254] [Help]
When i set onProgress function in ChatGPTClient, ,example:
<img width="809" alt="image" src="https://user-images.githubusercontent.com/39156049/228125584-ef28ca30-2986-4674-8d8a-79afa69cab60.png">

But，i can't get the details's usage in this prompt
<img width="610" alt="image" src="https://user-images.githubusercontent.com/39156049/228125700-b4d873a8-6037-4ca3-b1b0-88a9fc1b2ecd.png">

so, can everyone help me?

**Comment by waylaidwanderer** [comment]
OpenAI doesn't provide that information in streaming mode. You have to keep track of it yourself some other way.

**Reply by AutumnWhj** [reply]
May be i can use getTokenCountForMessage to count it , but it may not be accurate enough, so it's better to rely on the usage returned by OpenAI?

**Reply by waylaidwanderer** [reply]
Yeah for counting input that's the best option. I think for the output you can just count how many times `onProgress` is called with a token.
